src/security-council-mgmt/governors/SecurityCouncilMemberElectionGovernor.sol

```diff
@@ -90,31 +93,32 @@ contract SecurityCouncilMemberElectionGovernor is
         AddressUpgradeable.functionCallWithValue(target, data, value);
     }
 
-    /**
-     * internal/private state mutating functions *************
-     */
-
     /// @dev    `GovernorUpgradeable` function to execute a proposal overridden to handle nominee elections.
     ///         We know that _getTopNominees will return a full list of nominees because we checked it in _voteSucceeded.
     ///         Calls `SecurityCouncilManager.replaceCohort` with the list of nominees.
     function _execute(
         uint256 proposalId,
         address[] memory, /* targets */
         uint256[] memory, /* values */
-        bytes[] memory, /* calldatas */
+        bytes[] memory callDatas,
         bytes32 /* descriptionHash */
     ) internal override {
-        // we know that the list is full because we checked it in _voteSucceeded
+        // we know that the election index is part of the calldatas
+        uint256 electionIndex = ElectionGovernorLib.extractElectionIndex(callDatas);
+
+        // it's possible for this call to fail because of checks in the security council manager
+        // getting into a state inconsistent with the elections
+        // if it does then the DAO or the security council will need to take action to either
+        // remove this election (upgrade this contract and cancel the proposal), or update the
+        // manager state to be consistent with this election and allow the cohort to be replaced
+        // One of these actions should be taken otherwise the election will stay in this contract
+        // and could be executed at a later unintended date.
         securityCouncilManager.replaceCohort({
             _newCohort: topNominees(proposalId),
-            _cohort: nomineeElectionGovernor.currentCohort()
+            _cohort: nomineeElectionGovernor.electionIndexToCohort(electionIndex)
```

Comment:
should we put this in the new lib as well?

Reply:
Good catch! Updated